pyvcloud/vcd/org.py

```diff
@@ -546,3 +550,27 @@ def get_roles_query(self, name_filter=None):
             equality_filter=name_filter,
             qfilter=org_filter)
         return query, resource_type
+
+    def get_catalog_access_control_settings(self, catalog_name):
+        """
+        Get the access control settings of a catalog.
+        :param catalog_name: (str): The name of the catalog.
+        :return: Access control settings of the catalog.
+        """  # NOQA
+        catalog_resource = self.get_catalog(name=catalog_name)
+        control_access = self.client.get_linked_resource(
+            catalog_resource,
+            RelationType.DOWN,
+            EntityType.CONTROL_ACCESS_PARAMS.value)
+        access_settings = []
+        if hasattr(control_access, 'AccessSettings') and \
+           hasattr(control_access.AccessSettings, 'AccessSetting') and \
+           len(control_access.AccessSettings.AccessSetting) > 0:
+                        for access_setting in list(
+                                control_access.AccessSettings.AccessSetting):
+                            access_settings.append(access_settings_to_dict(
+                                access_setting))
+        result = to_dict(control_access)
+        if len(access_settings) > 0:
+            result['AccessSettings'] = access_settings
+        return result
```

Comment:
result can have None value and result is being accessed directly without None check in vcd-cli. Please handle this.

Reply:
result can never be None, because the to_dict function will map the minimum attributes from the control_access object. (ControlAccessParams will have the IsSharedToEveryone attribute even if there are no other access control setting information for the catalog.)